### DISCUSSION TITLE
View performance per account

### DISCUSSION THREAD

**Discussion by dickwolff** [ghostfolio/ghostfolio#1475] [Q&A] [answered]
I’m not sure if this is a feature (if so I haven’t found it yet), but is it possible to view performance graphs/percentages per account? 

So I have added about 4 accounts from different types of investments. I have an overall performance graphs, top gainers and losers etc, but this displays the details of all accounts combined. Can you also view these graphs per individual account? If not, is this something that would be developed in the future?

Thanks!

**Comment by dtslvr** [comment]
This is not possible at the moment, but definitely a desired feature.

It should work the very same way as the filtering on the allocations page.

The following endpoints need that filter functionality:

* `GET api/v1/portfolio/investments`
* `GET api/v1/portfolio/positions`
* `GET api/v2/portfolio/performance`

Inspiration: `GET api/v1/portfolio/details` (for allocations)
See `@Query('accounts') filterByAccounts?: string` and `@Query('assetClasses') filterByAssetClasses?: string`.

Let me know if you would like to work on (parts of) this and I can create proper issues.

**Reply by dickwolff** [reply]
Thanks, I might look into the backend code and see if it would be something I could work on!

**Reply by dtslvr** [reply]
Sure, I can approach the frontend integration after that if you'd like.

**Reply by dickwolff** [reply]
That would be nice.

I tried to run it locally today and after some arguing with Yarn it seems everything compiled and ran. However I got an error when I tried to add an activity, the API would throw an error about performance being undefined in the data provider service. I’ll try to look into it tomorrow as well, if I don’t find the issue I’ll contact you via Slack. Maybe you could give me some pointers.

**Reply by dtslvr** [reply]
Let's discuss it in Slack and if the error persists, we can create a ticket.

**Reply by vgibara** [reply]
Any update on this ? This is a featured i would really appreciate !

**Comment by dtslvr** [comment, marked as answer]
Hey @vgibara

This is actually possible: Go to _Portfolio_ > _Analysis_  and filter by an account.

**Reply by vgibara** [reply]
I can't get the information I want... I'd like to see the details or one portfolio with daily performance. 

I have this on Wealthsimple which is usefull to me.

![IMG_20230218_082130.jpg](https://user-images.githubusercontent.com/754856/219868261-8ec51d45-0c30-4b20-9496-ae090b6f9c2c.jpg)

**Reply by dickwolff** [reply]
When you go to the page suggested by Thomas and enter one of your accounts you can choose Today and then you will see the performance for today (however today is Saturday so it shows a bit weird)

![Screenshot_2023-02-18-14-24-58-68_e4424258c8b8649f6e67d283a50a2cbc](https://user-images.githubusercontent.com/5620002/219868324-3f79b7b1-dfd0-4d5a-a333-d2be0a5090fe.jpg)

**Comment by vgibara** [comment]
That's correct. Which is partly good, but I only see top movers. Not all of
my stocks tickers.

**Reply by dickwolff** [reply]
Fair point, that would be a nice addition. Not just for today, but YTD etc. Something like the Holdings list but instead of percent of portfolio show the gains for a given timeframe.

**Comment by vgibara** [comment]
Exactly !

% of portfolio is interesting, but could be mostly static in time, if not
much change in portfolio content. However, following value change over time
is interesting!

If I find a bit of time, I might check how to add this. Is the data already
available through the api ?

Vincent
